### PR TITLE
feat: Support Workspaces

### PR DESCRIPTION
**Pull request by HarryET**
Support workspaces with a feature to look for the cargo.toml in the directory above the project's

### CODE REVIEW

**Comment by lukaslueg** [comment]
Thanks for the PR!

This is, however, a too-naive implementation of Cargo's workspace handling. One would need to re-implement [`find_root()`](https://github.com/rust-lang/cargo/blob/b75281bee39818bd538f81dd8f82ec2fec059a39/src/cargo/core/workspace.rs#L594), taking care of excluded paths; I'd suggest a re-implementation since this is unlikely to change fundamentally, and I'd rather avoid a full-blown dependency on `cargo`.

I don't think we need a feature flag for this.

There need to be some tests for this.

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "built"
-version = "0.5.1"
+version = "0.5.2"
```
Comment (lukaslueg) [inline Cargo.toml:3]: Please don't bump the version in a PR, this is done separately.

File: src/lib.rs [hunk 2]
```diff
@@ -375,7 +375,13 @@ impl CIPlatform {
 
 fn get_build_deps(manifest_location: &path::Path) -> io::Result<Vec<(String, String)>> {
     let mut lock_buf = String::new();
-    fs::File::open(manifest_location.join("Cargo.lock"))?.read_to_string(&mut lock_buf)?;
+    let path: PathBuf;
+    if cfg!(feature = "workspace") {
+        path = manifest_location.join("../Cargo.lock")
```
Comment (lukaslueg) [inline src/lib.rs:380]: This is potentially platform-incompatible. There is `ancestors()` and `parent()` to handle this is a platform-agnostic way.

File: Cargo.toml (hunk 1 above)
Comment (HarryET) [reply Cargo.toml:3]: Sorry, I was using this as a git dependency for another project. Will change it back.

**Comment by HarryET** [comment]
> Thanks for the PR!
> 
> This is, however, a too-naive implementation of Cargo's workspace handling. One would need to re-implement [`find_root()`](https://github.com/rust-lang/cargo/blob/b75281bee39818bd538f81dd8f82ec2fec059a39/src/cargo/core/workspace.rs#L594), taking care of excluded paths; I'd suggest a re-implementation since this is unlikely to change fundamentally, and I'd rather avoid a full-blown dependency on `cargo`.
> 
> I don't think we need a feature flag for this.
> 
> There need to be some tests for this.

Thanks for the fast reply, the main reason I used this was because of another project I had needed it for, I don't particularly have time for working on it fully right now but just PRed incase it would be of use to the project. I'll keep this in mind and try revisit it when I get some more free time 😀

**Pull request closed without merging**